include/flatbuffers/code_generators.h

```diff
@@ -29,7 +29,7 @@ class BaseGenerator {
     EnsureDirExists(path.c_str());
     if (parser.opts.one_file) return path;
     std::string namespace_dir = path;  // Either empty or ends in separator.
-    auto &namespaces = ns.components;
+    auto namespaces = PrefixedNamespaceComponents(ns, parser.opts.lang);
```

Comment:
Ah yes, we can do that as required (not all languages use namespacing like Java). My comment was more about the use in Java, making sure we cover all cases.